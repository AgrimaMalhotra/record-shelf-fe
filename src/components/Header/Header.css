header {
    background-color: #38497E;
    color: white;
    height: 80px;
    display: flex;
    align-items: center;
    font-size: 45px;
    padding-left: 30px;
}

p {
    margin-top: 0px;
    margin-bottom: 0px;
    font-weight: lighter;
}

strong {
    font-family:
        Arial Black;
}