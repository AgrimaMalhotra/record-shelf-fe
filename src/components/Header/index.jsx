import React from 'react';
import { useNavigate } from 'react-router-dom';
import './Header.css';

const Header = () => {
  const navigate = useNavigate();
  const handleClick = () => {
    navigate('/');
  };
  return (
    <div>
      <header onClick={handleClick} data-testid="page-title">
        <p>
          My <strong>Record </strong> Shelf
        </p>
      </header>
    </div>
  );
};
export default Header;
