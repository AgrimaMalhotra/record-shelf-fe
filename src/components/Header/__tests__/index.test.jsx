import { fireEvent, render, screen } from '@testing-library/react';
import * as React from 'react';
import Header from '..';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

describe('Header', () => {
  it('should render correctly ', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
  it('should have clickable header', () => {
    render(<Header />);
    const clickableHeader = screen.getByTestId('page-title');
    fireEvent.click(clickableHeader);
    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });
});
