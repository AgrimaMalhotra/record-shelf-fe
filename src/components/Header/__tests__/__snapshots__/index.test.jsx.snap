// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly  1`] = `
<div>
  <div>
    <header
      data-testid="page-title"
    >
      <p>
        My 
        <strong>
          Record 
        </strong>
         Shelf
      </p>
    </header>
  </div>
</div>
`;
